src/libtiled/templatemanager.cpp

```diff
@@ -61,14 +65,28 @@ ObjectTemplate *TemplateManager::loadObjectTemplate(const QString &fileName, QSt
     if (!objectTemplate) {
         auto newTemplate = readObjectTemplate(fileName, error);
 
-        // This instance will not have an object. It is used to detect broken
-        // template references.
-        if (!newTemplate)
+        if (newTemplate) {
+            mWatcher->addPath(fileName);
+        } else {
+            // This instance will not have an object. It is used to detect broken
+            // template references.
             newTemplate = std::make_unique<ObjectTemplate>(fileName);
+        }
 
         objectTemplate = newTemplate.get();
         mObjectTemplates.insert(fileName, newTemplate.release());
     }
 
     return objectTemplate;
 }
+
+void TemplateManager::fileChanged(const QString &fileName)
+{
+    ObjectTemplate *objectTemplate = findObjectTemplate(fileName);
+
+    // Most likely the file was removed.
+    if (!objectTemplate)
+        return;
+
```

Comment:
So, the template file changed, and you found a previously loaded template, but then it's time to reload this template, right? I don't see this happening currently.

Probably needs something like:
```c++
auto newTemplate = readObjectTemplate(fileName);
if (newTemplate) {
    objectTemplate->setObject(newTemplate->object());
    emit objectTemplateChanged(objectTemplate);
}
```
In case or error we should probably report it using the `ERROR` function, which will show up in the Console and Issues views.

Reply:
Whoops! That was a bad oversight. I was too focused on making all the right integrations.